sympy/core/sympify.py

```diff
@@ -313,11 +324,26 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
             pass
 
     if not isinstance(a, string_types):
-        for coerce in (float, int):
-            try:
-                return sympify(coerce(a))
-            except (TypeError, ValueError, AttributeError, SympifyError):
-                continue
+        if _is_numpy_instance(a):
+            import numpy as np
+            if isinstance(a, np.number):
+                return sympify(a)
```

Comment:
Is this just calling sympify with the same input? How is this not infinite recursion?

Reply:
Oh, you're probably right.  But this was happening before wasn't it?

Reply:
What should we do in this case?

Reply:
To be honest, I can't figure out how to get to this code path. These types must be cased out earlier.